modules/objdetect/include/opencv2/objdetect/aruco_detector.hpp

```diff
@@ -388,6 +388,67 @@ CV_EXPORTS_W void drawDetectedMarkers(InputOutputArray image, InputArrayOfArrays
 CV_EXPORTS_W void generateImageMarker(const Dictionary &dictionary, int id, int sidePixels, OutputArray img,
                                       int borderBits = 1);
 
+struct CV_EXPORTS_W_SIMPLE RefineCharucoParameters {
+    RefineCharucoParameters() {
+        minMarkers = 3;
+        tryRefineMarkers = false;
+    }
+    Mat cameraMatrix; // cameraMatrix optional 3x3 floating-point camera matrix
+    Mat distCoeffs; // distCoeffs optional vector of distortion coefficients
+    int minMarkers; // minMarkers number of adjacent markers that must be detected to return a charuco corner, default = 3
+    bool tryRefineMarkers; // try to use refine board
+};
+
+class CV_EXPORTS_W CharucoDetector : public ArucoDetector {
+public:
+    /** @brief Basic CharucoDetector constructor
+     *
+     * @param board ChAruco board
+     * @param detectorParams marker detection parameters
+     * @param refineParams marker refine detection parameters
+     * @param charucoParams charuco detection parameters
+     */
+    CV_WRAP CharucoDetector(const Ptr<CharucoBoard> &board,
+                            const RefineCharucoParameters& refineCharucoParameters = RefineCharucoParameters(),
+                            const DetectorParameters &detectorParams = DetectorParameters(),
+                            const RefineParameters& refineParams = RefineParameters());
```

Comment:
Parameters names are controversial.
1. Place ARUCO part first and add it to parameters name.
2. Please ChaRuco refine parames last.

Reply:
if ARUCO part is implaced first, then default parameters isn't allowed.